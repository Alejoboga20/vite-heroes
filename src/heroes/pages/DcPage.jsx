import { HeroList } from '../components';

const publisher = 'DC Comics';

export const DcPage = () => {
	return (
		<>
			<h1>{publisher}</h1>
			<hr />

			<HeroList publisher={publisher} />
		</>
	);
};
