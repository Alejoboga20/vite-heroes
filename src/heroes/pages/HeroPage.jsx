import { Navigate, useNavigate, useParams } from 'react-router-dom';
import { getHeroById } from '../helpers';

export const HeroPage = () => {
	const { id } = useParams();
	const navigate = useNavigate();
	const hero = getHeroById(id);
	const { characters, alter_ego, publisher, first_appearance, superhero } = hero;

	const onNavigateBack = () => {
		navigate(-1);
	};

	if (!hero) return <Navigate to='/marvel' />;

	return (
		<div className='row mt-5'>
			<div className='col-4'>
				<img className='img-thumbnail' src={`/assets/heroes/${id}.jpg`} alt={superhero} />
			</div>

			<div className='col-8'>
				<h3>{superhero}</h3>
				<ul className='list-group list-group-flush'>
					<li className='list-group-item'>
						<b>Alter Ego:</b> {alter_ego}
					</li>
					<li className='list-group-item'>
						<b>Publisher:</b> {publisher}
					</li>
					<li className='list-group-item'>
						<b>First Appearance:</b> {first_appearance}
					</li>
				</ul>

				<h5 className='mt-3'>Characters</h5>
				<p>{characters}</p>

				<button className='btn btn-outline-primary' onClick={onNavigateBack}>
					Back
				</button>
			</div>
		</div>
	);
};
