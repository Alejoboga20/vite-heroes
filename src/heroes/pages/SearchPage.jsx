import { useLocation, useNavigate } from 'react-router-dom';
import queryString from 'query-string';

import { useForm } from '../../hooks';
import { HeroCard } from '../components';

export const SearchPage = () => {
	const navigate = useNavigate();
	const location = useLocation();

	const { q = '' } = queryString.parse(location.search);

	const { searchText, onInputChange } = useForm({
		searchText: '',
	});

	const onSearchSubmit = (event) => {
		event.preventDefault();

		if (searchText.trim() <= 1) return;

		navigate(`?q=${searchText}`);
	};

	return (
		<>
			<h1>SearchPage</h1>
			<hr />

			<div className='row'>
				<div className='col-5'>
					<h4>Searching</h4>
					<hr />

					<form onSubmit={onSearchSubmit}>
						<input
							type='text'
							placeholder='search hero'
							className='form-control'
							name='searchText'
							autoComplete='off'
							value={searchText}
							onChange={onInputChange}
						/>
					</form>

					<button className='btn btn-outline-primary mt-2' onClick={onSearchSubmit}>
						Search
					</button>
				</div>

				<div className='col-7'>
					<h4>Results</h4>
					<hr />

					<div className='alert alert-primary'>Search a hero</div>
					<div className='alert alert-danger'>
						There's no result with <b>{q}</b>
					</div>

					{/* <HeroCard /> */}
				</div>
			</div>
		</>
	);
};
