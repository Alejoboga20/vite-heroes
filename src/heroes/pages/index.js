export { HeroPage } from './HeroPage';
export { SearchPage } from './SearchPage';
export { DcPage } from './DcPage';
export { MarvelPage } from './MarvelPage';
