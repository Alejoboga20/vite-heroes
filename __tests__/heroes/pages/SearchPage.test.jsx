import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { SearchPage } from '../../../src/heroes/pages/SearchPage';

describe('SearchPage Tests', () => {
	test('should match Snapshot', () => {
		const { container } = render(
			<MemoryRouter>
				<SearchPage />
			</MemoryRouter>
		);

		expect(container).toMatchSnapshot();
	});

	test('should show result on search', () => {
		render(
			<MemoryRouter initialEntries={['/search?q=batman']}>
				<SearchPage />
			</MemoryRouter>
		);

		const inputValue = screen.getByRole('textbox');
		expect(inputValue.value).toBe('batman');

		const img = screen.getByRole('img');
		expect(img.src).toContain('/assets/heroes/dc-batman');

		const alert = screen.getByLabelText('alert-danger');
		expect(alert.style.display).toContain('none');
	});
});
