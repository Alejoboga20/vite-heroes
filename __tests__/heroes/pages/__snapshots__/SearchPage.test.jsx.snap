// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPage Tests should match Snapshot 1`] = `
<div>
  <h1>
    SearchPage
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="search hero"
          type="text"
          value=""
        />
      </form>
      <button
        class="btn btn-outline-primary mt-2"
      >
        Search
      </button>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="alert-danger"
        class="alert alert-primary"
      >
        Search a hero
      </div>
      <div
        class="alert alert-danger"
        style="display: none;"
      >
        There's no result with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
